src/browser/provider/built-in/dedicated/chrome/cdp.ts

```diff
@@ -1,228 +1,244 @@
 
-export async function createClient (runtimeInfo: RuntimeInfo): Promise<void> {
-    const { browserId, config, cdpPort } = runtimeInfo;
+        if (!config.headless)
+            await providerMethods.resizeLocalBrowserWindow(browserId, newWidth, newHeight, currentWidth, currentHeight);
 
-    let tab    = null;
-    let client = null;
+        viewportSize.width = newWidth;
+        viewportSize.height = newHeight;
 
-    try {
-        tab = await getActiveTab(cdpPort, browserId);
+        const client = await this.getActiveClient();
 
-        if (!tab)
-            return;
+        if (config.emulation) {
+            await this._setDeviceMetricsOverride(client, viewportSize.width, viewportSize.height, emulatedDevicePixelRatio, config.mobile);
 
-        client = await remoteChrome({ target: tab, port: cdpPort });
+            await client.Emulation.setVisibleSize({ width: viewportSize.width, height: viewportSize.height });
+        }
     }
-    catch (e) {
-        return;
+
+    public isHeadlessTab (): boolean {
+        return !!this._parentTarget && this._config.headless;
     }
 
-    runtimeInfo.tab    = tab;
-    runtimeInfo.client = client;
+    public async getActiveClient (): Promise<remoteChrome.ProtocolApi> {
+        const client = this._clients[this._clientKey];
 
-    await client.Page.enable();
-    await client.Network.enable({});
-    await client.Runtime.enable();
+        if (client)
+            return client;
 
-    await calculateEmulatedDevicePixelRatio(runtimeInfo);
+        return this._createClient();
+    }
 
-    if (config.emulation)
-        await setEmulation(runtimeInfo);
+    public async init (): Promise<void> {
+        try {
+            const tabs = await this._getTabs();
 
-    if (config.headless)
-        await enableDownloads(runtimeInfo);
-}
+            this._parentTarget = tabs.find(t => t.url.includes(this._runtimeInfo.browserId));
 
-export function isHeadlessTab ({ tab, config }: RuntimeInfo): boolean {
-    return tab && config.headless;
-}
+            if (!this._parentTarget)
+                return;
 
-export async function closeTab ({ tab, cdpPort }: RuntimeInfo): Promise<void> {
-    await remoteChrome.closeTab({ id: tab.id, port: cdpPort });
-}
+            const client = await this._createClient();
 
-export async function updateMobileViewportSize (runtimeInfo: RuntimeInfo): Promise<void> {
-    const windowDimensionsQueryResult = await runtimeInfo.client.Runtime.evaluate({
-        expression:    `(${GET_WINDOW_DIMENSIONS_INFO_SCRIPT})()`,
-        returnByValue: true
-    });
 
-    const windowDimensions = windowDimensionsQueryResult.result.value;
+            await this._calculateEmulatedDevicePixelRatio(client);
 
-    runtimeInfo.viewportSize.width  = windowDimensions.outerWidth;
-    runtimeInfo.viewportSize.height = windowDimensions.outerHeight;
-}
+            await this._setupClient(client);
+        }
+        catch (e) {
+            return;
```

Comment:
Can we handle this error in some way?

Reply:
I'm not sure that we need to handle this case. The try/catch should definitely be here to prevent further code execution. However, in the previous version, this situation is not handled as well. Probably @AndreyBelym can say something in detail?

Reply:
Ok. It's not related to this PR.
Try to find out more information from @AndreyBelym